lib/redisHandler.js

```diff
@@ -0,0 +1,231 @@
+const Resp = require('respjs');
+const EventEmitter = require('events');
+
+/**
+ * Class to handle a redis connection and provide events to react on for
+ * all incoming Redis commands
+ */
+class RedisHandler extends EventEmitter {
+    /**
+     * Initialize and register all data handlers to send out events on new commands
+     * @param socket Network Socket/Connection
+     * @param options options objects, currently mainly for logger
+     */
+    constructor(socket, options) {
+        super();
+
+        options = options || {};
+        this.log = options.log || console;
+        this.socket = socket;
+
+        this.sockedId = socket.remoteAddress + ':' + socket.remotePort;
+        this.initialized = false;
+
+        this.writeQueue = [];
+
+        this.resp = new Resp();
+
+        this.resp.on('error', (err) => {
+            this.log.error(this.sockedId + ' (Init=' + this.initialized + ') Redis error:' + err);
+            if (this.initialized) {
+                this.sendError(null,new Error('PARSER ERROR ' + err)); // TODO
+            }
+            else {
+                this.close();
+            }
+        });
+
+        this.resp.on('data', (data) => {
```

Comment:
If possible I would extract the anonymous method into a private method with a speaking name. Like `_handleCommand(data)` or something.